xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/DateRangePicker.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.test.ui.po;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.NotFoundException;
+import org.openqa.selenium.WebElement;
+
+/**
+ * Represents the actions possible on the date range picker widget.
+ *
+ * @version $Id$
+ * @since 14.9RC1
+ */
+public class DateRangePicker extends BaseElement
+{
+    private WebElement container;
+
+    /**
+     * Get the date range picker corresponding to the given element.
+     *
+     * @param element an element that has a date range picker assigned
+     */
+    public DateRangePicker(WebElement element)
+    {
+        this.container = (WebElement) ((JavascriptExecutor) getDriver()).executeScript(
```

Comment:
Is using javascript evaluation mandatory here?

Reply:
Yes, because the `daterangepicker` is a jquery data attribute and I cannot access it with selenium